### ISSUE TITLE
Rule Request: Discourage Equality Checks Against RawValue

### ISSUE THREAD

**Issue by Mazyod** [realm/SwiftLint#2146]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.
People are easily falling victims to equality check against `rawValue`, when it is much clearer/cleaner to check against the enum type.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.
```swift
// trigger
switch string {
case MyEnum.caseA.rawValue:
  break
default:
  break
}

if value == MyEnum.caseA.rawValue {
  ...
}

// non-trigger
let possibleValue = MyEnum(rawValue: value)
if possibleValue == .caseA {
  ..
}

switch possibleValue {
case .some(.caseA):
  break
}
```

3. Should the rule be configurable?
Probably not?

4. Should the rule be opt-in or enabled by default?
Opt-in, for sure, since it is extremely disruptive to existing codebases.
